state/testing.go

```diff
@@ -332,8 +332,14 @@ func testChangeAccountBalanceToZero(t *testing.T, buildPreState buildPreState) {
 	txn := newTxn(state, snap)
 	txn.SetBalance(addr1, big.NewInt(10))
 	txn.SetBalance(addr1, big.NewInt(0))
-	snap, _ = txn.Commit(true)
+	snap, _ = buildNewSnapshot(txn, snap, true)
 
 	txn = newTxn(state, snap)
 	assert.False(t, txn.Exist(addr1))
 }
+
+func buildNewSnapshot(txn *Txn, snapshot Snapshot, deleteEmptyObjects bool) (Snapshot, []byte) {
+	objs := txn.Commit(deleteEmptyObjects)
+
+	return txn.snapshot.Commit(objs)
```

Comment:
I mentioned today on grooming that you should avoid relying on `txn.snapshot` object to perform write operations because that can be simplified too in a followup PR. Besides, you are already passing a `snapshot` reference to the function that you are not using.

Is this something you can do?
```suggestion
	return snapshot.Commit(objs)
```
Also, I would not put this in a separate function, a one-liner with both things would work for me too.

Reply:
Yes, that was just copy paste error. I have removed function also